test_std_module/test.cpp

```diff
@@ -0,0 +1,30 @@
+// This test really only includes all the type_safe header and checks we can so a build
+// Some of these are commented out until we have a fix for a possible MS compiler bug in a later PR
+#include <type_safe/arithmetic_policy.hpp>
+#include <type_safe/boolean.hpp>
+#include <type_safe/bounded_type.hpp>
+//#include <type_safe/compact_optional.hpp>
+#include <type_safe/config.hpp>
+#include <type_safe/constrained_type.hpp>
+#include <type_safe/deferred_construction.hpp>
+#include <type_safe/downcast.hpp>
+#include <type_safe/flag.hpp>
+#include <type_safe/flag_set.hpp>
+#include <type_safe/floating_point.hpp>
+#include <type_safe/index.hpp>
+#include <type_safe/integer.hpp>
+#include <type_safe/narrow_cast.hpp>
+//#include <type_safe/optional.hpp>
+//#include <type_safe/optional_ref.hpp>
+#include <type_safe/output_parameter.hpp>
+#include <type_safe/reference.hpp>
+#include <type_safe/strong_typedef.hpp>
+#include <type_safe/tagged_union.hpp>
+#include <type_safe/types.hpp>
+//#include <type_safe/variant.hpp>
+//#include <type_safe/visitor.hpp>
+
+int main( )
+{
+    return 0;
+}
```

Comment:
Trailing newline.

Reply:
Ok just checked a PR in for the compiler bug, your suggestion worked.
I will rebase if you merge it and then fix all your comment. Sorry about the formating changes I try not to do that but must have missed them when reviewing.

I assume you mean do
target_compile_features(type_safe PRIVATE cxx_std_23)
I am assuming PRIVATE here because you don't want to force consumers, even the test programs, to use C++ 23.

Reply:
I think my check fixed all you requested changes